instrumentation/opentelemetry-instrumentation-openai/src/opentelemetry/instrumentation/openai/patch.py

```diff
@@ -0,0 +1,438 @@
+    if hasattr(result, "choices") and result.choices is not None:
+        responses = [
+            {
+                "role": (
+                    choice.message.role
+                    if choice.message and choice.message.role
+                    else "assistant"
+                ),
+                "content": extract_content(choice),
+                **(
+                    {"content_filter_results": choice["content_filter_results"]}
+                    if "content_filter_results" in choice
+                    else {}
+                ),
+            }
+            for choice in result.choices
+        ]
+        set_event_completion(span, responses)
+
+    if (
+        hasattr(result, "system_fingerprint")
+        and result.system_fingerprint is not None
+        and result.system_fingerprint != NOT_GIVEN
+    ):
+        set_span_attribute(
+            span,
+            SpanAttributes.LLM_SYSTEM_FINGERPRINT,
+            result.system_fingerprint,
+        )
+    # Get the usage
+    if hasattr(result, "usage") and result.usage is not None:
+        usage = result.usage
+        if usage is not None:
+            set_span_attribute(
+                span,
+                SpanAttributes.LLM_USAGE_PROMPT_TOKENS,
+                result.usage.prompt_tokens,
+            )
+            set_span_attribute(
+                span,
+                SpanAttributes.LLM_USAGE_COMPLETION_TOKENS,
+                result.usage.completion_tokens,
+            )
+            set_span_attribute(
+                span,
+                SpanAttributes.LLM_USAGE_TOTAL_TOKENS,
+                result.usage.total_tokens,
+            )
+
+
+def set_event_prompt(span: Span, prompt):
+    span.add_event(
+        name=SpanAttributes.LLM_CONTENT_PROMPT,
+        attributes={
+            SpanAttributes.LLM_PROMPTS: prompt,
+        },
+    )
+
+
+def set_span_attributes(span: Span, attributes: dict):
+    for field, value in attributes.model_dump(by_alias=True).items():
+        set_span_attribute(span, field, value)
+
+
+def set_event_completion(span: Span, result_content):
+    span.add_event(
+        name=SpanAttributes.LLM_CONTENT_COMPLETION,
+        attributes={
+            SpanAttributes.LLM_COMPLETIONS: json.dumps(result_content),
+        },
+    )
+
+
+def set_event_completion_chunk(span: Span, chunk):
+    span.add_event(
+        name=SpanAttributes.LLM_CONTENT_COMPLETION_CHUNK,
+        attributes={
+            SpanAttributes.LLM_CONTENT_COMPLETION_CHUNK: json.dumps(chunk),
+        },
+    )
+
+
+def set_span_attribute(span: Span, name, value):
+    if value is not None:
+        if value != "" or value != NOT_GIVEN:
+            if name == SpanAttributes.LLM_PROMPTS:
+                set_event_prompt(span, value)
+            else:
+                span.set_attribute(name, value)
+    return
+
+
+def is_streaming(kwargs):
+    return not (
+        kwargs.get("stream") is False
```

Comment:
Something like this may be a bit more readable?

```
return bool(kwargs.get("stream")) and kwargs.get("stream") != NOT_GIVEN
```

Reply:
this should be fine, as stream arguments are guaranteed to be always string or boolean as far as i know.